src/plugins/kibana/server/routes/api/ingest/register_pipelines.js

```diff
@@ -0,0 +1,54 @@
+import _ from 'lodash';
+import esToKibanaPipelineConverter from '../../../lib/converters/ingest_pipeline_api_es_to_kibana_converter';
+import handleESError from '../../../lib/handle_es_error';
+
+function handlePipelinesResponse(response) {
+  const pipelines = _.get(response, 'pipelines');
+  const results = _.map(pipelines, 'id');
+
+  return results;
+}
+
+function handlePipelineResponse(response) {
+  const esPipeline =  _.get(response, 'pipelines[0]');
+  const kibanaPipeline = esToKibanaPipelineConverter(esPipeline);
+  return kibanaPipeline;
+}
+
+export function registerPipelines(server) {
+  server.route({
+    path: '/api/kibana/ingest/pipelines',
+    method: 'GET',
+    handler: function (request, reply) {
+      const boundCallWithRequest = _.partial(server.plugins.elasticsearch.callWithRequest, request);
+
+      return boundCallWithRequest('transport.request', {
+        path: '/_ingest/pipeline/*',
+        method: 'GET'
+      })
+      .then(handlePipelinesResponse)
+      .then(reply)
+      .catch((error) => {
+        reply(handleESError(error));
+      });
+    }
+  });
+
+  server.route({
+    path: '/api/kibana/ingest/pipeline/{id}',
```

Comment:
For consistency this should be `pipelines` not `pipeline`. The idea is that you're requesting a single resource in the `pipelines` collection. It also avoids issues with client and server needing to have the exact same rules for pluralization.